src/shared-api-objects/current-user.ts

```diff
@@ -20,13 +21,29 @@ function getCurrentUser(
 ): Observable<LoggedInUser | UnauthenticatedUser> {
   return userSubject.asObservable().pipe(
     mergeAll(),
+    tap(setUserLanguage),
     map((r: LoggedInUserFetchResponse) =>
       opts.includeAuthStatus ? r.data : r.data.user
     ),
     filter(Boolean)
   ) as Observable<LoggedInUser | UnauthenticatedUser>;
 }
 
+function setUserLanguage(sessionResponse) {
+  if (sessionResponse?.data?.user?.userProperties?.defaultLocale) {
+    const locale = sessionResponse.data.user.userProperties.defaultLocale;
+    //@ts-ignore
+    const i18nInstance = i18n.default || i18n;
```

Comment:
Alternatively, what we could do here is set the `lang` attribute of `documentElement` (assuming this is our top prio - coming from the SSR, too).

Then just calling `i18nInstance.changeLanguage()` is sufficient. Is is using the detector logic again.

This way we always use the same detector logic / order.

Reply:
Additional remark:

In the outlined case we don't even need to trigger `i18nInstance.changeLanguage()`  here. When setting up the `i18nInstance` we could install a mutation observer on the `documentElement` (no subtree), which triggers when attributes change.

In this case we would always be consistent.

Reply:
Remark:
For the empty `lng` argument of `changeLanguage` see https://github.com/i18next/i18next/blob/master/src/i18next.js#L270.

Reply:
> For the empty `lng` argument of `changeLanguage` see https://github.com/i18next/i18next/blob/master/src/i18next.js#L270.

I was going to ask where you found that! Definitely not in the docs. Good find.

Reply:
Using this and a mutation observer on the `documentElement` makes sense to me. I'll make the change.

Reply:
Awesome! Thanks :+1:!